crates/core/src/client/api.rs

```diff
@@ -6,59 +6,58 @@ use reth_rpc_types::{
     TransactionReceipt,
 };
 use starknet::core::types::{BlockId as StarknetBlockId, BroadcastedInvokeTransactionV1, FieldElement};
-use starknet::providers::jsonrpc::JsonRpcTransport;
-use starknet::providers::JsonRpcClient;
+use starknet::providers::Provider;
 
 use super::errors::EthApiError;
 use crate::models::balance::TokenBalances;
 use crate::models::transaction::StarknetTransactions;
 
 #[async_trait]
-pub trait KakarotEthApi<T: JsonRpcTransport>: KakarotStarknetApi<T> {
-    async fn block_number(&self) -> Result<U64, EthApiError<T::Error>>;
+pub trait KakarotEthApi<P: Provider + Send + Sync>: KakarotStarknetApi<P> + Send + Sync {
```

Comment:
Why do we need "P: Provider + Send + Sync" and then "<P> + Send + Sync"

Reply:
Good question, somehow we need each one to be able to be passed around threads